cellular_configs/v1alpha1/commoncoreranconfig_types.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CommonCoreRANStatus defines the observed state of CommonCoreRAN
+type CommonCoreRANConfigStatus struct {
+}
+
+// +kubebuilder:object:root=true
+type CommonCoreRANConfig struct {
```

Comment:
this name is not good. Is it not better to split this in 2 CRD(s)
1 for PLMN
1 for Communication Configuration.

It means we end up with potentially many CR(s), but it is composable.

other option is to keep It all in 1 CRD and use CommonConfig or just Config.
RANCore is very odd in my view

Reply:
@henderiw I called it CommonCoreRANConfig because the PLMN/NSSAI part is common for RAN also I took the parameters that are common among all the core NFs. 

Right now I only proposed CRD specific to AMF, but later we can have specific ones for SMF, UPF and other NFs if needed. 

But If you want I can change it to `CommonCoreConfig` and let the RAN to have its own parameters. 

@josephthaliath do you have an opinion here?

Reply:
@arora-sagar, @henderiw , If we think only PLMN Info is common between RAN and Core , then we can create a seperate CRD for this and name it PLMNInfo itself. But if we think we will add more parameters  in future which are common between RAN and Core, then a CRD with name CommonConfig can be used as placeholder for such parameters. Domething like this:
1) CommonConfig ( common across RAN and Core  NFs) 
2) CommonCoreConfig ( common across Core NFs )
3) CommonRANConfig (common across all RAN NFs )

Reply:
So I am of the opinion we need only 1 CRD that embeds the other smaller CRD that define a unit of x (e.g. PLMN is such a small unit).

So to depict it we call this config.cellular.config.nephio.org. it is the meta CR that is applied to the cluster and embeds these smaller self contained elements. As such we don't need to care about RAN or Core. We apply it to the NF deployment through the parameterRef. So no need for RAN or Core. Just it is applied to the NF of choice. You parse the CR units from there after you picked up the CR.

Now the only issue might be CR size but we can solve this by adding multiple configs with a different name.
Thoughts ?